src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1033,7 +1019,7 @@ public IEnumerable<Document> GetDocumentsFrom(DocumentsOperationContext context,
             if (collectionName == null)
                 yield break;
 
-            var table = context.Transaction.InnerTransaction.OpenTable(DocsSchema,
```

Comment:
As I understand the issue was that we passed "blindly" `DocsSchema` without checking if a collection is compressed or not, is that correct? 

What will happen in a scenario where a collection was compressed and then the compression was disabled for it. Then we'll pass `DocsSchema`. How is that case supposed to work?

Reply:
For reads, this doesn't matter, we do the actual operation on a document by document scenario.
We can revert this safely.

Reply:
@ArieSLV ?

Reply:
> As I understand the issue was that we passed "blindly" DocsSchema without checking if a collection is compressed or not, is that correct?

Correct

> What will happen in a scenario where a collection was compressed and then the compression was disabled for it. Then we'll pass DocsSchema. How is that case supposed to work?

As Oren previously mentioned, the schema is important at the time of writing. For entries placed in a small section, we set the flag Compressed per entry: https://github.com/ravendb/ravendb/blob/1bae88adfcc53300d50e36561320c1e53a3246e0/src/Voron/Data/RawData/RawDataSection.cs#L229C24-L229C24

In cases where an entry does not fit in the small section, we use a large one and apply the Compressed flag to the page.

During reading, we use these flags regardless of the current schema settings.

Reply:
Thank you. It's clear to me now.